components/script/dom/path2d.rs

```diff
@@ -0,0 +1,315 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::cell::RefCell;
+
+use canvas_traits::canvas::PathSegment;
+use dom_struct::dom_struct;
+use js::rust::HandleObject;
+use script_bindings::str::DOMString;
+use svgtypes::SimplifyingPathParser;
+
+use crate::dom::bindings::codegen::Bindings::CanvasRenderingContext2DBinding::Path2DMethods;
+use crate::dom::bindings::error::{Error, Fallible};
+use crate::dom::bindings::reflector::{Reflector, reflect_dom_object_with_proto};
+use crate::dom::bindings::root::DomRoot;
+use crate::dom::globalscope::GlobalScope;
+use crate::script_runtime::CanGc;
+
+#[derive(Clone, Copy, JSTraceable)]
+struct PathSegmentWrapper(#[no_trace = "Does not contain managed objects"] PathSegment);
+malloc_size_of::malloc_size_of_is_0!(PathSegmentWrapper);
```

Comment:
Why is this wrapper type necessary?

Reply:
Because PathSegmentWrapper is contained in a `#[dom_struct]`, so it needs to implement JSTraceable. JSTraceable is not available in the crate defining PathSegment, so the wrapper is needed.

Reply:
right but you could use `#[no_trace]` on the struct field in `Path2D` instead of inside the wrapper type.